sdk-ext-ui/src/androidTest/java/cash/z/ecc/sdk/ext/ui/regex/ZecRegexTest.kt

```diff
@@ -0,0 +1,62 @@
+package cash.z.ecc.sdk.ext.ui.regex
+
+import androidx.test.filters.MediumTest
+import cash.z.ecc.sdk.ext.ui.R
+import cash.z.ecc.sdk.ext.ui.test.getStringResourceWithArgs
+import cash.z.ecc.sdk.model.MonetarySeparators
+import org.junit.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+class ZecRegexTest {
+
+    @Test
+    @MediumTest
+    fun check_regex_validity() {
+        val separators = MonetarySeparators.current()
+        val regexString = getStringResourceWithArgs(
+            R.string.zec_amount_regex_filter,
+            arrayOf(
+                separators.grouping,
+                separators.grouping,
+                separators.decimal,
+                separators.decimal
+            )
+        )
+        assertNotNull(regexString)
+
+        val regexAmountChecker = regexString.toRegex()
+
+        regexAmountChecker.also {
+            assertNotNull(regexAmountChecker)
+            assertTrue(regexAmountChecker.pattern.isNotEmpty())
+        }
+
+        regexAmountChecker.also {
+            // valid inputs
+            assertTrue(it.matches(""))
+            assertTrue(it.matches("123"))
+            assertTrue(it.matches("123${separators.grouping}"))
+            assertTrue(it.matches("123${separators.grouping}456"))
+            assertTrue(it.matches("123${separators.decimal}"))
+            assertTrue(it.matches("123${separators.decimal}456"))
+            assertTrue(it.matches("123${separators.grouping}456${separators.decimal}"))
+            assertTrue(it.matches("123${separators.grouping}456${separators.decimal}789"))
```

Comment:
I'd like to see an additional case with 2 groupings.  E.g. 1,234,567.00

Reply:
That is something which needs additional regex enhancement, thank you for pointing it out.